### PR TITLE
Add signature of Timeout

### PR DESCRIPTION
**Pull request by masahito1997**
This PR add signature of Timeout module.

NOTE: Timeout module has sub module(ex: `Timeout::Error` ). 
But, I first wrote in rbs. So, this is first step and doesn't include the sub module.

### CODE REVIEW

File: test/stdlib/Timeout_test.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    assert_send_type  "(::BigDecimal sec) { (::BigDecimal sec) -> untyped } -> untyped",
+                      Timeout, :timeout, BigDecimal("1.123456789123456789"), &proc
+
+    hard_process = Proc.new { _calc_pi }
+    refute_send_type  "(::Numeric sec) { (::Numeric sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 0.001, &hard_process
+    refute_send_type  "(::Numeric sec, singleton(Exception) klass) { (::Numeric sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 0.001, TimeoutTestException, &hard_process
+    refute_send_type  "(::Numeric sec, singleton(Exception) klass, String message) { (::Numeric sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 0.001, TimeoutTestException, "timeout test error", &hard_process
```
Comment (pocke) [inline test/stdlib/Timeout_test.rb:30]: I'm not sure that this `refute_send_type` usage is appropriate. 

I guess `refute_send_type` is designed to check an unacceptable type. For example, `Timeout.timeout('1')` is invalid with the type, so checking `Timeout.timeout('1')` call with `refute_send_type` is the designed usage.

But these tests, such as `Timeout.timeout(0.001){...}`, is acceptable with the type. It just checks `Timeout.timeout` raises an error, but the raised error is an "expected" error. So the `refute_send_type` calls just suppress the error. I think `refute_send_type` is not appropriate in this case.


related: #588

File: stdlib/timeout/0/timeout.rbs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  # The exception thrown to terminate the given block cannot be rescued inside the
+  # block unless `klass` is given explicitly. However, the block can use ensure to
+  # prevent the handling of the exception.  For that reason, this method cannot be
+  # relied on to enforce timeouts for untrusted blocks.
+  #
+  # Note that this is both a method of module Timeout, so you can `include
+  # Timeout` into your classes so they have a #timeout method, as well as a module
+  # method, so you can call it directly as Timeout.timeout().
+  #
+  def self.timeout: (Numeric? sec, ?singleton(Exception) klass, ?String message) { (Numeric sec) -> untyped } -> untyped
```
[the remark this answers is not in the document]
Comment (masahito1997) [reply stdlib/timeout/0/timeout.rbs]: Thank you for your advice and suggestion.
I committed it.

File: test/stdlib/Timeout_test.rb (hunk 1 above)
Comment (masahito1997) [reply test/stdlib/Timeout_test.rb]: Thank you for your advice! I can understand.
I fixed it in 083beb9 . Would you please review again.

File: stdlib/timeout/0/timeout.rbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (masahito1997) [reply stdlib/timeout/0/timeout.rbs]: Thank you for your recommendations.
I fixed it in 5ff07e8213672b1788eac4b70057c42cab05d8c1

**Comment by pocke** [comment]
Thanks for your contribution! :clap:  I merged this pull request with a small fix https://github.com/ruby/rbs/pull/586/commits/6119e0d31851b157d9c7b992cb15ff582b62dbf0.

**Pull request merged**